packages/merge-styles/src/concatStyleSets.ts

```diff
@@ -1,30 +1,203 @@
+ * @param styleSet1 The first style set to be concatenated.
+ * @param styleSet2 The second style set to be concatenated.
+ * @param styleSet3 The third style set to be concatenated.
+ * @param styleSet4 The fourth style set to be concatenated.
+ */
+export function concatStyleSets<
+  TStyleSet1 extends IStyleSet<TStyleSet1>,
+  TStyleSet2 extends IStyleSet<TStyleSet2>,
+  TStyleSet3 extends IStyleSet<TStyleSet3>,
+  TStyleSet4 extends IStyleSet<TStyleSet4>
+>(
+  styleSet1: TStyleSet1 | false | null | undefined,
+  styleSet2: TStyleSet2 | false | null | undefined,
+  styleSet3: TStyleSet3 | false | null | undefined,
+  styleSet4: TStyleSet4 | false | null | undefined
+): IConcatenatedStyleSet<TStyleSet1 & TStyleSet2 & TStyleSet3 & TStyleSet4>;
+
+/**
+ * Combine a set of styles together (but does not register css classes).
+ * @param styleSet1 The first style set to be concatenated.
+ * @param styleSet2 The second style set to be concatenated.
+ * @param styleSet3 The third style set to be concatenated.
+ * @param styleSet4 The fourth style set to be concatenated.
+ * @param styleSet5 The fifth set to be concatenated.
+ */
+export function concatStyleSets<
+  TStyleSet1 extends IStyleSet<TStyleSet1>,
+  TStyleSet2 extends IStyleSet<TStyleSet2>,
+  TStyleSet3 extends IStyleSet<TStyleSet3>,
+  TStyleSet4 extends IStyleSet<TStyleSet4>,
+  TStyleSet5 extends IStyleSet<TStyleSet5>
+>(
+  styleSet1: TStyleSet1 | false | null | undefined,
+  styleSet2: TStyleSet2 | false | null | undefined,
+  styleSet3: TStyleSet3 | false | null | undefined,
+  styleSet4: TStyleSet4 | false | null | undefined,
+  styleSet5: TStyleSet5 | false | null | undefined
+): IConcatenatedStyleSet<TStyleSet1 & TStyleSet2 & TStyleSet3 & TStyleSet4 & TStyleSet5>;
+
+/**
+ * Combine a set of styles together (but does not register css classes).
+ * @param styleSet1 The first style set to be concatenated.
+ * @param styleSet2 The second style set to be concatenated.
+ * @param styleSet3 The third style set to be concatenated.
+ * @param styleSet4 The fourth style set to be concatenated.
+ * @param styleSet5 The fifth set to be concatenated.
+ * @param styleSet6 The sixth set to be concatenated.
+ */
+export function concatStyleSets<
+  TStyleSet1 extends IStyleSet<TStyleSet1>,
+  TStyleSet2 extends IStyleSet<TStyleSet2>,
+  TStyleSet3 extends IStyleSet<TStyleSet3>,
+  TStyleSet4 extends IStyleSet<TStyleSet4>,
+  TStyleSet5 extends IStyleSet<TStyleSet5>,
+  TStyleSet6 extends IStyleSet<TStyleSet6>
+>(
+  styleSet1: TStyleSet1 | false | null | undefined,
+  styleSet2: TStyleSet2 | false | null | undefined,
+  styleSet3: TStyleSet3 | false | null | undefined,
+  styleSet4: TStyleSet4 | false | null | undefined,
+  styleSet5: TStyleSet5 | false | null | undefined,
+  styleSet6: TStyleSet6 | false | null | undefined
+): IConcatenatedStyleSet<TStyleSet1 & TStyleSet2 & TStyleSet3 & TStyleSet4 & TStyleSet5 & TStyleSet6>;
+
+/**
```

Comment:
Don't know if my eyes are playing tricks on me but is this the same as the implementation below?

Reply:
Yes, it is the same.